drivers/SmartThings/zigbee-switch/src/multi-switch-no-master/init.lua

```diff
@@ -37,35 +51,97 @@ local function is_multi_switch_no_master(opts, driver, device)
         return true
       end
   end
-
   return false
 end
 
-local function component_to_endpoint(device, component_id)
-  local ep_num = component_id:match("switch(%d)")
-  return ep_num and tonumber(ep_num) + 1 or 1
+local function get_children_amount(device)
+  for _, fingerprint in ipairs(MULTI_SWITCH_NO_MASTER_FINGERPRINTS) do
+    if device:get_model() == fingerprint.model then
+      return fingerprint.children
+    end
+  end
+end
+
+local function device_added(driver, device, event)
+  if device.network_type == st_device.NETWORK_TYPE_ZIGBEE then
+    local children_amount = get_children_amount(device)
+    for i = 1, children_amount do
+      local name = string.format("%s %d", device.label, i)
+      local metadata = {
+        type = "EDGE_CHILD",
+        label = name,
+        profile = "multi-switch-no-master",
+        parent_device_id = device.id,
+        parent_assigned_child_key = string.format("%02X", i),
+        vendor_provided_label = name,
+      }
+      driver:try_create_device(metadata)
+    end
+  end
 end
 
-local function endpoint_to_component(device, ep)
-  local switch_comp = string.format("switch%d", ep - 1)
-  if device.profile.components[switch_comp] ~= nil then
-    return switch_comp
-  else
-    return "main"
+local function find_child(parent, ep_id)
+  return parent:get_child_by_parent_assigned_key(string.format("%02X", ep_id))
+end
+
+local function device_init(driver, device, event)
+  if device.network_type == st_device.NETWORK_TYPE_ZIGBEE then
+    device:set_find_child(find_child)
   end
 end
 
-local device_init = function(self, device)
-  device:set_component_to_endpoint_fn(component_to_endpoint)
-  device:set_endpoint_to_component_fn(endpoint_to_component)
+local function switch_on_command_handler(driver, device, command)
+  device:send(OnOff.server.commands.On(device))
+
 end
 
+local function switch_off_command_handler(driver, device, command)
+  device:send(OnOff.server.commands.Off(device))
+end
+
+local function on_off_command_handler(driver, device, value, zb_rx)
+  local event
+  
+  if value == OnOff.server.commands.On.ID then
+    event = capabilities.switch.switch.on()
+  elseif value == OnOff.server.commands.Off.ID then
+    event = capabilities.switch.switch.off()
+  end
+  
+  if event ~= nil then
+    device:emit_event(event)
+  end
+end
+
+local function do_refresh(self, device)
+  device:send(OnOff.attributes.OnOff:read(device))
+end
+  
 local multi_switch_no_master = {
   NAME = "multi switch no master",
+  zigbee_handlers = {
+    cluster = {
+      [OnOff.ID] = {
+        [OnOff.server.commands.On.ID] = on_off_command_handler,
+        [OnOff.server.commands.Off.ID] = on_off_command_handler
+      }
+    }
```

Comment:
you need to be handling an attribute report here, not the command

Reply:
> you need to be handling an attribute report here, not the command

I managed to eliminate those methods, now it is handled by defaults. The issue was in mapping endpoints